roles/edge-image-builder/tasks/main.yml

```diff
@@ -44,58 +71,148 @@
   block:
   - name: Create network directory
     file:
-      path: "{{ eib_path }}/network"
+      path: "{{ eib_path }}/metal3/network"
       state: directory
   - name: Copy configure-network.sh
     copy:
+      mode: 0755
       src: "configure-network.sh"
-      dest: "{{ eib_path }}/network"
+      dest: "{{ eib_path }}/metal3/network"
 
-- name: Build EIB container image from git checkout
-  when: eib_git_version != ""
-  block:
-  - name: Git checkout
-    ansible.builtin.git:
-      repo: 'https://github.com/suse-edge/edge-image-builder.git'
-      dest: "{{ working_dir }}/edge-image-builder"
-      version: "{{ eib_git_version }}"
+- name: Run EIB container to generate the metal3 image
+  become: true
+  containers.podman.podman_container:
+    name: eib
+    rm: true
+    image: "{{ eib_container_image }}"
+    volumes:
+      - "{{ eib_path }}/metal3:/eib"
+    detach: false
+    state: started
+    force_restart: no
+    privileged: true
+    command: "build --definition-file config-eib.yaml"
 
-  - name: build the EIB container image
-    become: true
-    containers.podman.podman_image:
-      name: eib:dev
-      path: "{{ working_dir }}/edge-image-builder"
+- name: Generate sha256sum for EIB image
+  become: true
+  shell: |
+    sha256sum {{ eib_metal3_output_image_name }} > {{ eib_metal3_output_image_name }}.sha256
+  args:
+    chdir: "{{ eib_path }}/metal3"
 
-  - name: set fact to use the local eib:dev image
-    set_fact:
-      eib_container_image: "localhost/eib:dev"
+- name: set fact for use in the example-manifests role
+  set_fact:
+    downstream_cluster_image: "{{ eib_metal3_output_image_name }}"
 
-- name: Run EIB container to generate the custom image
+- name: "Ensure {{ image_cache_dir }} exists"
+  become: true
```

Comment:
same

Reply:
`image_cache_dir` defaults to:

```
libvirt_images_dir: /var/lib/libvirt/images
image_cache_dir: "{{ libvirt_images_dir }}/image-cache"
```
So because `/var/lib/libvirt/images` is root owned, become is required - an alternative could be to make that dir writeable for the current user, or to use a different dir for the image cache - in either case I don't think we want to create a world-writeable cache directory though, any preference on the approach here?

Reply:
It is ok to keep it with become, thanks!


Just in case, as part of the hypervisor setup I usually like to do the following:

```
usermod -a -G libvirt,qemu suse

# I think this is not needed, need to check.
#cat << EOF >> /etc/libvirt/libvirtd.conf
#unix_sock_group = "libvirt"
#unix_sock_rw_perms = "0770"
#auth_unix_rw = "none"
#EOF

cat << EOF >> /etc/libvirt/libvirt.conf
uri_default = "qemu:///system"
EOF

systemctl restart libvirtd

cat << EOF > /etc/polkit-1/rules.d/10-grant-libvirt
polkit.addRule(function(action, subject) {
  if (action.id == "org.libvirt.unix.manage" && subject.isInGroup("libvirt")) {
    return polkit.Result.YES;
  }
});
EOF

chmod 0771 /var/lib/libvirt/images
chgrp libvirt /var/lib/libvirt/images
```

Reply:
Thanks @e-minguez - I had issues getting non-default storage pools to work with policykit so ended up [disabling the qemu security driver instead](https://github.com/suse-edge/metal3-demo/blob/main/roles/libvirt-setup/tasks/libvirt_service_tasks.yml#L15) - perhaps your rule addition could avoid that, will give it a try and potentially align in a follow-up PR 👍 

I'll see if I can use group permissions to reduce the become usage to only the EIB build tasks :)

Reply:
Just in case, those tweaks were performed on a SLES host, so I'm not sure if all of them will apply directly (or if they will work at all) on Leap 😅

Reply:
Ok this is more complicated that I thought, because we can't add the user to the `libvirt` group, then have that group reflected in the current shell.

I think that leaves two possible options:

- Move from a localhost connection to SSH so we can use [reset_connection](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/meta_module.html) to get a new shell via SSH
- Move the image-cache to a different directory (for example under $HOME)

I think the second option is the simplest, @e-minguez do you have any experience of solving this, or is the hypervisor user/group pre-configured in your environment?

Reply:
I'm starting to think just keep become as it is the easiest way 😄 

I've setup those servers manually so I didn't have those issues or I could just use `newgrp` (or `exec newgrp <group>` to avoid replacing the shell).

Maybe a dirty solution is to try to run the exec newgrp command here https://github.com/suse-edge/metal3-demo/blob/main/roles/libvirt-setup/tasks/libvirt_service_tasks.yml#L23 ?

Anyway, let's keep the become for now I guess.

On an offtopic note, TIL about `sg` https://www.jeffgeerling.com/comment/17549#comment-17549

Reply:
Ok I moved the image-cache directory under $HOME which removed most of the root tasks (the podman containers still need to be root for now as mentioned elsewhere)